### PR TITLE
delete endpoint for tokens

### LINKED ISSUE #3258: Add endpoint to delete token from service account
**User Story**
Add a new endpoint to delete given token from the service account

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: add an endpoint to delete token from the service account

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3258 

```release-note
a new endpoint to delete tokens from the service accounts: DELETE /api/v1/projects/{project_id}/serviceaccounts/{serviceaccount_id}/tokens/{token_id}
```

### CODE REVIEW

File: api/pkg/handler/v1/serviceaccount/token.go [hunk 1]
```diff
@@ -172,6 +172,30 @@ func PatchTokenEndpoint(projectProvider provider.ProjectProvider, serviceAccount
...
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		_, err = serviceAccountProvider.Get(userInfo, req.ServiceAccountID, &provider.ServiceAccountGetOptions{RemovePrefix: false})
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		return nil, serviceAccountTokenProvider.Delete(userInfo, req.TokenID)
```
Comment (p0lyn0mial) [inline api/pkg/handler/v1/serviceaccount/token.go]: wrap the error, use `common.KubernetesErrorToHTTPError` method.

File: api/pkg/handler/v1/serviceaccount/token.go [hunk 2]
```diff
@@ -172,6 +172,30 @@ func PatchTokenEndpoint(projectProvider provider.ProjectProvider, serviceAccount
...
+		if err != nil {
+			return nil, errors.NewBadRequest(err.Error())
+		}
+
+		_, err = projectProvider.Get(userInfo, req.ProjectID, &provider.ProjectGetOptions{})
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		_, err = serviceAccountProvider.Get(userInfo, req.ServiceAccountID, &provider.ServiceAccountGetOptions{RemovePrefix: false})
```
Comment (p0lyn0mial) [inline api/pkg/handler/v1/serviceaccount/token.go]: nit: you don't need `&provider.ServiceAccountGetOptions{RemovePrefix: false}`

File: api/pkg/handler/v1/serviceaccount/token_test.go [hunk 3]
```diff
@@ -532,6 +534,71 @@ func TestUpdateToken(t *testing.T) {
 	}
 }
 
+func TestDeleteToken(t *testing.T) {
```
Comment (p0lyn0mial) [inline api/pkg/handler/v1/serviceaccount/token_test.go:537]: please validate the rsp from the API

File: api/pkg/handler/v1/serviceaccount/token.go (hunk 1 above)
Comment (zreigz) [reply api/pkg/handler/v1/serviceaccount/token.go]: done

File: api/pkg/handler/v1/serviceaccount/token_test.go (hunk 3 above)
Comment (zreigz) [reply api/pkg/handler/v1/serviceaccount/token_test.go:537]: done

File: api/pkg/handler/v1/serviceaccount/token.go (hunk 2 above)
Comment (zreigz) [reply api/pkg/handler/v1/serviceaccount/token.go]: done

**Comment by zreigz** [comment]
@p0lyn0mial PTAL

**Pull request merged**